components/memory/test/test_flash.c

```diff
@@ -0,0 +1,27 @@
+// Copyright 2022 PWr in Space, Krzysztof Gliwiński
+
+#include <string.h>
+
+#include "esp_log.h"
+#include "flash.h"
+#include "unity.h"
+
+#define FILE_NAME (const char*)"/spiffs/test_data.txt"
+#define EXPECTED_DATA \
+  (const char*)"Test text, nothing to find here...\nWidzisz mie?\0"
+
+#define TAG "Flash test"
+
+TEST_CASE("Flash init and read file test", "[flash]") {
+  TEST_ASSERT_EQUAL(FLASH_OK, FLASH_init(1));
+}
+
+TEST_CASE("Flash read file test", "[flash") {
+  char readData[40];
+  FlashResult res;
+  res = FLASH_read_all_data(FILE_NAME, readData, 1000000);
```

Comment:
zamiast 1000000, sizeof(readData), ma to przeciwdziałać wywaleniu poza pamiec

Reply:
I (4935) SPIFFS: Used size 1004, total size 956561
zostaje 1000000 bo ta libka chyba coś innego sprawdza, nie rozmiar samego pliku, albo coś robię nie tak xD
W każdym razie powinno być g a nie jest